packages/opentelemetry-core/src/internal/validators.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const VALID_KEY_REGEX = /^[a-z][_0-9a-z-*/]{0,255}$/;
+const VALID_VALUE_REGEX = /^[\x20-\x2b\x2d-\x3c\x3e-\x7e]{0,255}[\x21-\x2b\x2d-\x3c\x3e-\x7e]$/;
```

Comment:
What would you think about making this a little more compact and readable (also less bytes of code for the browser)?

Specifically: 
- We can use negated lookahead to check for the "not comma or equals" criteria and then use the whole character range as explained in the comment
- `'\x20'` is `' '`, `'\x21'` is `'!'`, and `'\x7e'` is `'~'` and replacing those allows simplifying the range

The resulting regex is `/^((?![,=])[ -~]){0,255}(?![,=])[!-~]$/`, which seems to work for our test cases, see https://regex101.com/r/MGRHOq/1

Reply:
I did a little more reading and there is some [guidance around not using lookaheads](https://stackoverflow.com/questions/1843459/javascript-the-good-parts-why-is-lookahead-not-good) because it could be bugging in IE or reduce performance. I don't know enough about whether that is truly the case, but if nothing else, they can be confusing at first glance as well.

What if we simply had two regexes? So:
```
const VALID_VALUE_BASE = /^[ -~]{0, 255}[!-~]$/;
const INVALID_VALUE_COMMA_EQUAL = /,|=/;

export function validateValue(value: string): boolean {
  return VALID_VALUE_BASE.test(value) && !INVALID_VALUE_COMMA_EQUAL.test(value);
}
```

Reply:
Thanks for sharing, makes sense to me. The same has been incorporated in the PR with extra test cases, PTAL.